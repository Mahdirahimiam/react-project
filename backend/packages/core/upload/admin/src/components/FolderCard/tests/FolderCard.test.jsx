import React from 'react';

import { Flex, lightTheme, ThemeProvider, Typography } from '@strapi/design-system';
import { act, fireEvent, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { FolderCard } from '../FolderCard';
import { FolderCardBody } from '../FolderCardBody';
import { FolderCardBodyAction } from '../FolderCardBodyAction';
import { FolderCardCheckbox } from '../FolderCardCheckbox';

const ID_FIXTURE = 'folder';

// eslint-disable-next-line react/prop-types
const ComponentFixture = ({ to, ...props }) => {
  return (
    <ThemeProvider theme={lightTheme}>
      <MemoryRouter>
        <FolderCard
          id={ID_FIXTURE}
          ariaLabel="Folder 1"
          startAction={<></>}
          onClick={() => {}}
          to={to}
          {...props}
        >
          <FolderCardBody as="h2">
            <FolderCardBodyAction onClick={() => {}} to={to}>
              <Flex direction="column" alignItems="flex-start">
                <Typography variant="omega" fontWeight="semiBold">
                  Pictures
                </Typography>
              </Flex>
            </FolderCardBodyAction>
          </FolderCardBody>
        </FolderCard>
      </MemoryRouter>
    </ThemeProvider>
  );
};

const setup = (props) => render(<ComponentFixture {...props} />);

describe('FolderCard', () => {
  test('renders and matches the snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('properly calls the onClick callback', () => {
    const callback = jest.fn();
    const { container } = setup({ onClick: callback });

    act(() => {
      fireEvent(container.querySelector('button'), new MouseEvent('click', { bubbles: true }));
    });

    expect(callback).toHaveBeenCalledTimes(1);
  });

  test('has all required ids set when rendering a start action', () => {
    const { container } = setup({
      startAction: <FolderCardCheckbox value={false} />,
    });

    expect(container).toMatchSnapshot();
    expect(container.querySelector(`[id="${ID_FIXTURE}-3-title"]`)).toBeInTheDocument();
    expect(
      container.querySelector(`[aria-labelledby="${ID_FIXTURE}-3-title"]`)
    ).toBeInTheDocument();
  });

  test('renders as a link with to prop', () => {
    const { container } = setup({
      to: '/michka-page',
    });

    const link = container.querySelector('a[aria-hidden=true]');
    expect(link).toHaveAttribute('href', '/michka-page');
  });

  test('renders as a button without to prop', () => {
    const { container } = setup();

    expect(container.querySelector('button[aria-hidden=true]')).toBeInTheDocument();
  });
});
