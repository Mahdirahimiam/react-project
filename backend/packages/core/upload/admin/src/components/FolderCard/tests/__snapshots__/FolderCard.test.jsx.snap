// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderCard has all required ids set when rendering a start action 1`] = `
.c19 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c13 {
  font-size: 0.875rem;
  line-height: 1.43;
  font-weight: 500;
  color: #32324d;
}

.c0 {
  position: relative;
}

.c2 {
  background: #ffffff;
  padding: 12px;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  border-color: #eaeaef;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  cursor: pointer;
}

.c4 {
  position: relative;
  z-index: 2;
}

.c6 {
  background: #eaf5ff;
  color: #66b7f1;
  padding-top: 8px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 12px;
  border-radius: 4px;
}

.c8 {
  position: relative;
  overflow: hidden;
  max-width: 100%;
}

.c11 {
  padding: 4px;
  max-width: 100%;
}

.c15 {
  right: 16px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c9 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c16 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 {
  height: 18px;
  min-width: 18px;
  margin: 0;
  border-radius: 4px;
  border: 1px solid #c0c0cf;
  -webkit-appearance: none;
  background-color: #ffffff;
  cursor: pointer;
}

.c5:checked {
  background-color: #4945ff;
  border: 1px solid #4945ff;
}

.c5:checked:after {
  content: '';
  display: block;
  position: relative;
  background: url("data:image/svg+xml,%3csvg%20width='10'%20height='8'%20viewBox='0%200%2010%208'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M8.55323%200.396973C8.63135%200.316355%208.76051%200.315811%208.83931%200.395768L9.86256%201.43407C9.93893%201.51157%209.93935%201.6359%209.86349%201.7139L4.06401%207.67724C3.9859%207.75755%203.85707%207.75805%203.77834%207.67834L0.13866%203.99333C0.0617798%203.91549%200.0617102%203.79032%200.138504%203.7124L1.16213%202.67372C1.24038%202.59432%201.36843%202.59422%201.4468%202.67348L3.92174%205.17647L8.55323%200.396973Z'%20fill='white'%20/%3e%3c/svg%3e") no-repeat no-repeat center center;
  width: 10px;
  height: 10px;
  left: 50%;
  top: 50%;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

.c5:checked:disabled:after {
  background: url("data:image/svg+xml,%3csvg%20width='10'%20height='8'%20viewBox='0%200%2010%208'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M8.55323%200.396973C8.63135%200.316355%208.76051%200.315811%208.83931%200.395768L9.86256%201.43407C9.93893%201.51157%209.93935%201.6359%209.86349%201.7139L4.06401%207.67724C3.9859%207.75755%203.85707%207.75805%203.77834%207.67834L0.13866%203.99333C0.0617798%203.91549%200.0617102%203.79032%200.138504%203.7124L1.16213%202.67372C1.24038%202.59432%201.36843%202.59422%201.4468%202.67348L3.92174%205.17647L8.55323%200.396973Z'%20fill='%238E8EA9'%20/%3e%3c/svg%3e") no-repeat no-repeat center center;
}

.c5:disabled {
  background-color: #dcdce4;
  border: 1px solid #c0c0cf;
}

.c5:indeterminate {
  background-color: #4945ff;
  border: 1px solid #4945ff;
}

.c5:indeterminate:after {
  content: '';
  display: block;
  position: relative;
  color: white;
  height: 2px;
  width: 10px;
  background-color: #ffffff;
  left: 50%;
  top: 50%;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

.c5:indeterminate:disabled {
  background-color: #dcdce4;
  border: 1px solid #c0c0cf;
}

.c5:indeterminate:disabled:after {
  background-color: #8e8ea9;
}

.c17 > * {
  margin-left: 0;
  margin-right: 0;
}

.c17 > * + * {
  margin-left: 8px;
}

.c18 {
  position: absolute;
  top: 12px;
}

.c1 {
  height: 100%;
  left: 0;
  position: absolute;
  opacity: 0;
  top: 0;
  width: 100%;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 path {
  fill: currentColor;
}

.c14 {
  display: none;
}

.c10 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c12:focus {
  outline: 2px solid #4945ff;
  outline-offset: -2px;
}

<div>
  <div
    class="c0 "
    tabindex="0"
  >
    <button
      aria-hidden="true"
      aria-label="Folder 1"
      class="c1"
      tabindex="-1"
      type="button"
    />
    <div
      class="c2 c3"
    >
      <div
        class="c4"
      >
        <div
          class=""
        >
          <input
            aria-labelledby="folder-3-title"
            class="c5"
            type="checkbox"
          />
        </div>
      </div>
      <div
        class="c6"
      >
        <svg
          class="c7"
          fill="none"
          height="1.5rem"
          viewBox="0 0 24 24"
          width="1.5rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.414 5H21a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h7.414l2 2Z"
            fill="#212134"
          />
        </svg>
      </div>
      <h2
        class="c8 c9 c10"
        id="folder-3-title"
        overflow="hidden"
      >
        <button
          class="c11 c12"
          type="button"
        >
          <div
            class="c9"
          >
            <span
              class="c13"
            >
              Pictures
            </span>
          </div>
        </button>
      </h2>
      <div
        class="c14"
      >
        <div
          class="c15 c16 c17 c18"
        />
      </div>
    </div>
  </div>
  <div
    class="c19"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`FolderCard renders and matches the snapshot 1`] = `
.c17 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c11 {
  font-size: 0.875rem;
  line-height: 1.43;
  font-weight: 500;
  color: #32324d;
}

.c0 {
  position: relative;
}

.c2 {
  background: #ffffff;
  padding: 12px;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  border-color: #eaeaef;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  cursor: pointer;
}

.c4 {
  background: #eaf5ff;
  color: #66b7f1;
  padding-top: 8px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 12px;
  border-radius: 4px;
}

.c6 {
  position: relative;
  overflow: hidden;
  max-width: 100%;
}

.c9 {
  padding: 4px;
  max-width: 100%;
}

.c13 {
  right: 16px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c7 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c15 > * {
  margin-left: 0;
  margin-right: 0;
}

.c15 > * + * {
  margin-left: 8px;
}

.c16 {
  position: absolute;
  top: 12px;
}

.c1 {
  height: 100%;
  left: 0;
  position: absolute;
  opacity: 0;
  top: 0;
  width: 100%;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 path {
  fill: currentColor;
}

.c12 {
  display: none;
}

.c8 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10:focus {
  outline: 2px solid #4945ff;
  outline-offset: -2px;
}

<div>
  <div
    class="c0 "
    tabindex="0"
  >
    <button
      aria-hidden="true"
      aria-label="Folder 1"
      class="c1"
      tabindex="-1"
      type="button"
    />
    <div
      class="c2 c3"
    >
      <div
        class="c4"
      >
        <svg
          class="c5"
          fill="none"
          height="1.5rem"
          viewBox="0 0 24 24"
          width="1.5rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.414 5H21a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h7.414l2 2Z"
            fill="#212134"
          />
        </svg>
      </div>
      <h2
        class="c6 c7 c8"
        id="folder-1-title"
        overflow="hidden"
      >
        <button
          class="c9 c10"
          type="button"
        >
          <div
            class="c7"
          >
            <span
              class="c11"
            >
              Pictures
            </span>
          </div>
        </button>
      </h2>
      <div
        class="c12"
      >
        <div
          class="c13 c14 c15 c16"
        />
      </div>
    </div>
  </div>
  <div
    class="c17"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
